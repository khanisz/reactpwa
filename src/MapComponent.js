import React, { Component } from "react";
import GoogleMapReact from "google-map-react";

import MyGreatPlace from './my_great_place.jsx';

class MapComponent extends Component {
  static defaultProps = {
    center: {
      lat: 51.11,
      lng: 17.02
    },
    zoom: 14
  };

  render() {
    return (
      <div style={{ height: "95vh", width: "100%" }}>
        <GoogleMapReact
          bootstrapURLKeys={{ key: "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ" }}
          defaultCenter={this.props.center}
          defaultZoom={this.props.zoom}
        >
        
          <MyGreatPlace lat={51.109306} lng={17.030001} text={'A'}/>
          <MyGreatPlace lat={51.107756} lng={17.027516} text={'B'}/>
          <MyGreatPlace lat={51.111528} lng={17.038073} text={'C'}/>
        </GoogleMapReact>
      </div>
    );
  }
}

export default MapComponent;
