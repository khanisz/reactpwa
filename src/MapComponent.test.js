import React from "react";
import { shallow } from "enzyme";
import MapComponent from "./MapComponent";

describe("MapComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MapComponent debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<MapComponent />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 1 map Point", () => {
    const mapPoint = [
      {
        name: "1",
        lat: 51.109306,
        lng: 17.030001
      }
    ];
    const component = shallow(<MapComponent mapPoint={mapPoint} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 2 map Points", () => {
    const mapPoint = [
      {
        name: "1",
        lat: 51.109306,
        lng: 17.030001
      },
      {
        name: "2",
        lat: 51.107756,
        lng: 17.027516
      }
    ];
    const component = shallow(<MapComponent mapPoint={mapPoint} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 3 map Points", () => {
    const mapPoint = [
      {
        name: "1",
        lat: 51.109306,
        lng: 17.030001
      },
      {
        name: "2",
        lat: 51.107756,
        lng: 17.027516
      },
      {
        name: "3",
        lat: 51.111528,
        lng: 17.038073
      }
    ];
    const component = shallow(<MapComponent mapPoint={mapPoint} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 0 map Points", () => {
    const mapPoint = [];
    const component = shallow(<MapComponent mapPoint={mapPoint} />);
    expect(component).toMatchSnapshot();
  });
});
