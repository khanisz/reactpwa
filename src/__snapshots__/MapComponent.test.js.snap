// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent should render correctly in "debug" mode 1`] = `
<div
  style={
    Object {
      "height": "91vh",
      "width": "100%",
    }
  }
>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.110479,
        "lng": 17.033256,
      }
    }
    defaultZoom={14}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    ...Loading
  </GoogleMap>
</div>
`;

exports[`MapComponent should render correctly with 0 map Points 1`] = `
<div
  style={
    Object {
      "height": "91vh",
      "width": "100%",
    }
  }
>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.110479,
        "lng": 17.033256,
      }
    }
    defaultZoom={14}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  />
</div>
`;

exports[`MapComponent should render correctly with 1 map Point 1`] = `
<div
  style={
    Object {
      "height": "91vh",
      "width": "100%",
    }
  }
>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.110479,
        "lng": 17.033256,
      }
    }
    defaultZoom={14}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    <MyGreatPlace
      lat={51.109306}
      lng={17.030001}
      text="1"
    />
  </GoogleMap>
</div>
`;

exports[`MapComponent should render correctly with 2 map Points 1`] = `
<div
  style={
    Object {
      "height": "91vh",
      "width": "100%",
    }
  }
>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.110479,
        "lng": 17.033256,
      }
    }
    defaultZoom={14}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    <MyGreatPlace
      lat={51.109306}
      lng={17.030001}
      text="1"
    />
    <MyGreatPlace
      lat={51.107756}
      lng={17.027516}
      text="2"
    />
  </GoogleMap>
</div>
`;

exports[`MapComponent should render correctly with 3 map Points 1`] = `
<div
  style={
    Object {
      "height": "91vh",
      "width": "100%",
    }
  }
>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.110479,
        "lng": 17.033256,
      }
    }
    defaultZoom={14}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    <MyGreatPlace
      lat={51.109306}
      lng={17.030001}
      text="1"
    />
    <MyGreatPlace
      lat={51.107756}
      lng={17.027516}
      text="2"
    />
    <MyGreatPlace
      lat={51.111528}
      lng={17.038073}
      text="3"
    />
  </GoogleMap>
</div>
`;

exports[`MapComponent should render correctly with no props 1`] = `
<div
  style={
    Object {
      "height": "91vh",
      "width": "100%",
    }
  }
>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyAN_7EdSgJsXIE-2z87T-mz3hIDH-S9GjQ",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.110479,
        "lng": 17.033256,
      }
    }
    defaultZoom={14}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    ...Loading
  </GoogleMap>
</div>
`;
